qiskit_dynamics/pulse_simulator/pulse_simulator.py

```diff
@@ -0,0 +1,526 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Pulse-enabled simulator backend.
+"""
+
+import datetime
+import uuid
+
+from typing import List, Optional, Union
+import copy
+import numpy as np
+from scipy.integrate._ivp.ivp import OdeResult  # pylint: disable=unused-import
+
+from qiskit import pulse
+from qiskit.qobj.utils import MeasLevel
+from qiskit.qobj.common import QobjHeader
+from qiskit.transpiler import Target
+from qiskit.pulse import Schedule, ScheduleBlock
+from qiskit.pulse.transforms.canonicalization import block_to_schedule
+from qiskit.providers import Provider
+from qiskit.providers.options import Options
+from qiskit.providers.backend import BackendV2
+from qiskit.result import Result
+from qiskit.result.models import ExperimentResult, ExperimentResultData
+
+from qiskit import QiskitError, QuantumCircuit
+from qiskit import schedule as build_schedule
+from qiskit.quantum_info import Statevector, DensityMatrix
+
+from qiskit_dynamics.solvers.solver_classes import Solver
+
+from .dynamics_job import DynamicsJob
+from .pulse_simulator_utils import (
+    _get_dressed_state_decomposition,
+    _get_lab_frame_static_hamiltonian,
+    _get_memory_slot_probabilities,
+    _sample_probability_dict,
+    _get_counts_from_samples,
+)
+
+
+class PulseSimulator(BackendV2):
```

Comment:
Should we call it `DynamicSimulator`, `ScheduleSimulator`, or something else, instead of `PulseSimulator`?

Reply:
Maybe `DynamicsBackend` would be good?